packages/integration-sdk-core/src/data/createIntegrationRelationship.ts

```diff
@@ -265,7 +265,9 @@ function createRelationship({
     _type,
     _class: relationshipClass,
     _fromEntityKey: fromKey,
+    _fromType: fromType,
```

Comment:
Unfortunately, this will cause persister errors, which enforces that only known properties with leading underscores (`_<prop>`) are accepted. Any unknown properties are rejected.

```
22:10:24.820Z ERROR local: Error occurred while synchronizing collected data (integrationInstanceId=58d07123-2eef-4cd6-940c-2974d482222d, synchronizationJobId=7c63088c-87cf-4431-b7ef-663a4a5e3d02, integrationJobId=7c63088c-87cf-4431-b7ef-663a4a5e3d02)
  Error: Error uploading collected data (API response: code=RELATIONSHIP_UPLOAD_FAILED_VALIDATION, message="2 validation error(s):
  - Error in relationships[0]._fromType (reason=NOT_ALLOWED)
  - Error in relationships[0]._toType (reason=NOT_ALLOWED)
  ").
```

We'll need to rethink the approach here. Maybe you could load entity `_key` and `_type` into a map as you load entities, and fetch `_type` by `_key` as you load relationships?

```diff
+ const entityKeyToTypeMap = new Map<string, string>();
  async function handleGraphObjectFile(parsedData: FlushedGraphObjectData) {
    if (parsedData.entities) {
+     for (const { _key, _type } of parsedData.entities) entityKeyToTypeMap.set(_key, _type);
      await store.addEntities(parsedData.entities);
    }
    if (parsedData.relationships) {
      await store.addRelationships(
-       parsedData.relationships
+       parsedData.relationships.map((r) => ({
+         ...r,
+         _toType: entityKeyToTypeMap.get(r._toEntityKey),
+         _fromType: entityKeyToTypeMap.get(r._fromEntityKey),
+       }))
      );
    }
  }
```

This would work decently if you enforce that both entities are always sent to Neo4j before the relationship, but I'm not sure exactly the sequence of things here.

Reply:
Thanks for the catch Nick! 

So is it just that I chose names that were preceded by an underscore?  Because I don't believe there's any firm requirement for the Neo4j upload functionality that they be named that way.  Would simply removing the underscore and using `fromType` and `toType` and treating them like normal optional properties be an acceptable alternative?

Reply:
Also, was the above error you listed from running a sync command?  I can be sure to run that test against any future changes to this code (and maybe add a unit test for it too).

Reply:
After a bit more testing, it seems that it's probably not ideal to remove the underscore and have it flagged as something other than metadata.  I've removed collection of _fromType and _toType in the latest comit.